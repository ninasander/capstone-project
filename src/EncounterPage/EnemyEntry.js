import React from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

EnemyEntry.propTypes = {
  enemy: PropTypes.string.isRequired,
  ac: PropTypes.number.isRequired,
  maxHP: PropTypes.number.isRequired,
}

export default function EnemyEntry({ enemy, ac, maxHP }) {
  return (
    <EnemyEntryStyled>
      <h1>{enemy}</h1>
      <p>HP: {maxHP}</p>
      <p>AC: {ac}</p>
    </EnemyEntryStyled>
  )
}

const EnemyEntryStyled = styled.div`
  margin: 20px;
  padding: 20px;
  display: grid;
  grid-template-columns: 1fr 1fr;
  background: hsla(208, 40%, 28%, 1);
  background: linear-gradient(
      180deg,
      hsla(208, 40%, 28%, 1) 0%,
      hsla(208, 40%, 23%, 1) 51%,
      hsla(207, 38%, 18%, 1) 100%
    )
    fixed no-repeat;
  color: white;
  border-radius: 5px;
  box-shadow: 4px 4px 6px rgba(8, 15, 21, 0.3);
  h1 {
    grid-column: span 2;
    margin: 0;
    font-size: 140%;
  }
  p {
    margin: 20px 0 0;
  }
`
