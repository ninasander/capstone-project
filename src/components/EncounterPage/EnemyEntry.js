import React, { useState } from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import HPCalculator from './HPCalculator'

EnemyEntry.propTypes = {
  enemyName: PropTypes.string.isRequired,
  armorClass: PropTypes.string.isRequired,
  HP: PropTypes.string.isRequired,
}

export default function EnemyEntry({ enemyName, armorClass, HP }) {
  const [currentHP, setCurrentHP] = useState(HP)

  return (
    <EnemyEntryStyled>
      <h2>{enemyName}</h2>
      <p>
        HP: {currentHP}/{HP}
      </p>
      <HPCalculator currentHP={currentHP} HP={HP} setCurrentHP={setCurrentHP} />
      <p>AC: {armorClass}</p>
    </EnemyEntryStyled>
  )
}

const EnemyEntryStyled = styled.div`
  margin: 20px;
  padding: 20px;
  display: grid;
  grid-gap: 20px;
  justify-content: space-between;
  grid-template-columns: repeat(3, 1fr);
  background: hsla(208, 40%, 28%, 1);
  background: linear-gradient(
      180deg,
      hsla(208, 40%, 28%, 1) 0%,
      hsla(208, 40%, 23%, 1) 51%,
      hsla(207, 38%, 18%, 1) 100%
    )
    fixed no-repeat;
  color: white;
  border-radius: 5px;
  box-shadow: 4px 4px 6px rgba(8, 15, 21, 0.3);

  h2 {
    grid-column: span 3;
    margin: 0;
    font-size: 140%;
  }
  p {
    margin: 0;
  }
`
