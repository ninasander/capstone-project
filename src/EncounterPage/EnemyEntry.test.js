import React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import EnemyEntry from '../EncounterPage/EnemyEntry'
import 'jest-styled-components'
import currentHP from './EnemyEntry'

describe('EnemyEntry', () => {
  const enemy = 'Enemy 1'
  const maxHP = 50
  const HP = 50
  const ac = 17

  it('displays the enemy name', () => {
    const { getByText } = render(
      <EnemyEntry enemy={enemy} maxHP={maxHP} ac={ac} HP={HP} />
    )

    expect(getByText(enemy)).toBeInTheDocument()
  })
  it('displays the current and max HP of the enemy', () => {
    const HPEntry = 'HP: ' + HP + '/' + maxHP
    const { getByText } = render(
      <EnemyEntry enemy={enemy} maxHP={maxHP} HP={HP} ac={ac} />
    )

    expect(getByText(HPEntry)).toBeInTheDocument()
  })
  it('displays the enemy AC', () => {
    const acEntry = 'AC: ' + ac
    const { getByText } = render(
      <EnemyEntry enemy={enemy} maxHP={maxHP} ac={ac} />
    )

    expect(getByText(acEntry)).toBeInTheDocument()
  })

  it('renders correctly', () => {
    const tree = renderer.create(
      <EnemyEntry enemy={enemy} maxHP={maxHP} ac={ac} />
    )
    expect(tree).toMatchSnapshot()
  })
})
