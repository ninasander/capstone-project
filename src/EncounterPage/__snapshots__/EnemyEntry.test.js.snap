// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnemyEntry renders correctly 1`] = `
.c0 {
  margin: 20px;
  padding: 20px;
  display: grid;
  grid-template-columns: 1fr 1fr;
  background: hsla(208,40%,28%,1);
  background: linear-gradient( 180deg,hsla(208,40%,28%,1) 0%,hsla(208,40%,23%,1) 51%,hsla(207,38%,18%,1) 100% ) fixed no-repeat;
  color: white;
  border-radius: 5px;
  box-shadow: 4px 4px 6px rgba(8,15,21,0.3);
}

.c0 h2 {
  grid-column: span 2;
  margin: 0;
}

.c0 p {
  margin: 20px 0 0;
}

<div
  className="c0"
>
  <h2>
    Enemy 1
  </h2>
  <p>
    HP: 
    50
  </p>
  <p>
    AC: 
    17
  </p>
</div>
`;
